lua/roslyn/commands.lua

```diff
@@ -5,109 +5,125 @@ local M = {}
-    stop = {
-        impl = function()
-            local client = vim.lsp.get_clients({ name = "roslyn" })[1]
-            if not client then
-                return
-            end
-
-            -- TODO: Change this to `client:request` when minimal version is `0.11`
-            ---@diagnostic disable-next-line: missing-parameter, param-type-mismatch
-            client.stop(true)
-        end,
-    },
-    target = {
-        impl = function()
-            local bufnr = vim.api.nvim_get_current_buf()
-            local root = vim.b.roslyn_root or require("roslyn.sln.utils").root(bufnr)
-
-            local roslyn_lsp = require("roslyn.lsp")
-
-            local targets = vim.iter({ root.solutions, root.solution_filters }):flatten():totable()
-            vim.ui.select(targets or {}, { prompt = "Select target solution: " }, function(file)
-                if not file then
-                    return
-                end
-
-                vim.lsp.stop_client(vim.lsp.get_clients({ name = "roslyn" }), true)
-                local sln_dir = vim.fs.dirname(file)
-                roslyn_lsp.start(bufnr, assert(sln_dir), roslyn_lsp.on_init_sln(file))
-            end)
-        end,
-    },
+	restart = {
+		impl = function()
+			local client = vim.lsp.get_clients({ name = "roslyn" })[1]
+			if not client then
+				return
+			end
+
+			local attached_buffers = vim.tbl_keys(client.attached_buffers)
+
+			---@type function | nil
+			local remove_listener = nil
+
+			local function restart_lsp()
+				for _, buffer in ipairs(attached_buffers) do
+					if vim.api.nvim_buf_is_valid(buffer) then
+						vim.api.nvim_exec_autocmds("FileType", { group = "Roslyn", buffer = buffer })
+					end
+				end
+				if remove_listener then
+					remove_listener()
+				end
+			end
+
+			remove_listener = roslyn_emitter:on("stopped", restart_lsp)
+
+			local force_stop = vim.loop.os_uname().sysname == "Windows_NT"
+			client.stop(force_stop)
+		end,
+	},
+	stop = {
+		impl = function()
+			local client = vim.lsp.get_clients({ name = "roslyn" })[1]
+			if not client then
+				return
+			end
+
+			-- TODO: Change this to `client:request` when minimal version is `0.11`
+			---@diagnostic disable-next-line: missing-parameter, param-type-mismatch
+			client.stop(true)
+		end,
+	},
+	target = {
+		impl = function()
+			select_sln_and_start_lsp()
+		end,
+	},
+	start = {
+		impl = function()
+			select_sln_and_start_lsp()
+		end
+	},
 }
 
 ---@param opts table
 ---@see vim.api.nvim_create_user_command
 local function roslyn(opts)
-    local fargs = opts.fargs
-    local cmd = fargs[1]
-    local args = #fargs > 1 and vim.list_slice(fargs, 2, #fargs) or {}
-    local subcommand = subcommand_tbl[cmd]
-    if type(subcommand) == "table" and type(subcommand.impl) == "function" then
-        subcommand.impl(args, opts)
-        return
-    end
-
-    vim.notify(cmd_name .. ": Unknown subcommand: " .. cmd, vim.log.levels.ERROR, { title = "roslyn.nvim" })
+	local fargs = opts.fargs
```

Comment:
Seems like this is formatted with tabs instead of spaces which I have configured in .stylua.toml. Please check the conform config to see if it is correctly finding the config file for stylua. It might also be that you are using a different formatter.